### PR TITLE
Test work

### PR DESCRIPTION
**Pull request by pravinpoudel**
test code of mongo db issue resolved

### CODE REVIEW

File: backend/db/mongo.go [hunk 1]
```diff
@@ -25,14 +25,17 @@ type Mongo struct {
...
+	}
+
 	gc := db.database.Collection("games")
 	res, err := gc.DeleteOne(context.Background(), bson.M{
-		"name":         game.Name,
-		"author":       game.Developer,
-		"creationdate": game.CreationDate,
-		"version":      game.Version,
-		"tags":         game.Tags,
+		"_id": primitiveObjectId,
```
Comment (kp-121) [inline backend/db/mongo.go:36]: Good

File: backend/db/database_test.go [hunk 2]
```diff
@@ -32,7 +32,25 @@ var (
 		Developer:    "tester",
 		CreationDate: time.Date(1900, 1, 2, 0, 0, 0, 0, time.UTC),
 		Version:      "0.0.1",
-		Tags:         []string{"tag1", "tag2"},
+		Tags:         []string{"tag001", "tag011"},
+	}
+)
+
+var (
```
Comment (kp-121) [inline backend/db/database_test.go]: Why do we need more mock games?

File: backend/db/database_test.go [hunk 3]
```diff
@@ -61,14 +80,14 @@ func TestDatabase_GameID(t *testing.T) {
...
+	id1, _ := _db.AddGame(game3)
 
-	game0.Id = id0
-	game1.Id = id1
+	game2.Id = id0
+	game3.Id = id1
 
-	res0, _ := _db.GetGamesByTags([]string{"tag0"}, false)
-	res1, _ := _db.GetGamesByTags([]string{"tag1"}, false)
+	res0, _ := _db.GetGamesByTags([]string{"tag100"}, false)
```
Comment (kp-121) [inline backend/db/database_test.go]: Why did we change which games are tested?

**Comment by kp-121** [comment]
I don't think the database_test.go is up to date. Have you pulled main into your branch?

**Review by kp-121 — CHANGES_REQUESTED** [review]

File: backend/db/database_test.go (hunk 3 above)
Comment (pravinpoudel) [reply backend/db/database_test.go]: ohh i am sorry the delete function was not working so i made 4 total game object to test and now i realised that only two is enough so don't need to change the previous game object. Delete was not working which i realised late, i will change it and push it

File: backend/db/database_test.go [hunk 4]
```diff
@@ -67,8 +68,8 @@ func TestDatabase_Tags(t *testing.T) {
 	game0.Id = id0
 	game1.Id = id1
 
-	res0, _ := _db.GetGamesByTags([]string{"tag0"}, false)
-	res1, _ := _db.GetGamesByTags([]string{"tag1"}, false)
+	res0, _ := _db.GetGamesByTags([]string{"tag000"}, false)
```
Comment (kp-121) [inline backend/db/database_test.go]: These changes aren't adding anything to the tests

**Review by kp-121 — CHANGES_REQUESTED** [review]

**Review by kp-121 — APPROVED** [review]

**Pull request merged**